data/frame.go

```diff
@@ -344,6 +345,151 @@ func (f *Field) Nullable() bool {
 	return f.Type().Nullable()
 }
 
+// FloatAt returns a float64 at the specified index idx.
+// It will panic if idx is out of range.
```

Comment:
Since method returns error, couldn't you return an error instead of panic? As an alternative, skip returning error and panic for "unsupported field type" as well since that is the only error returned.

Reply:
I think generally it makes sense to leave checking length to the caller, as the check can generally be done just once logically somewhere in the function that uses this.

In terms of returning an error. As per the documentation an error can be returned in the string case: ```// If the Field type is a string, then strconv.ParseFloat is called on it and will return
// an error if ParseFloat errors.``` But would want to keep the error anyways in case we add Field Types that would make no sense to convert to a float.